src/libpriv/rpmostree-core.c

```diff
@@ -552,24 +562,32 @@ rpmostree_context_setup (RpmOstreeContext    *self,
                          GCancellable  *cancellable,
                          GError       **error)
 {
+  const char *releasever = NULL;
   g_autofree char **enabled_repos = NULL;
   g_autofree char **instlangs = NULL;
+  /* This exists (as a canonically empty dir) at least on RHEL7+ */
+  static const char emptydir_path[] = "/usr/share/empty";
 
   self->spec = g_object_ref (spec);
 
-  if (install_root)
-    dnf_context_set_install_root (self->hifctx, install_root);
-  else
+  g_variant_dict_lookup (self->spec->dict, "releasever", "&s", &releasever);
+
+  if (!install_root)
+    install_root = emptydir_path;
+  if (!source_root)
     {
-      /* We need this to ensure that libdnf doesn't read the host system packages */
-      if (!rpmostree_context_ensure_tmpdir (self, "dummy-instroot", error))
-        return FALSE;
-      g_autofree char *root = g_strconcat (self->tmpdir_path, "/dummy-instroot", NULL);
-      dnf_context_set_install_root (self->hifctx, root);
+      source_root = emptydir_path;
+      /* Hackaround libdnf's auto-introspection semantics. For e.g.
+       * treecompose/container, we currently require using .repo files which
+       * don't reference $releasever.
+       */
+      dnf_context_set_release_ver (self->hifctx, releasever ?: "rpmostree-unset-releasver");
     }
+  else if (releasever)
```

Comment:
Hmm, this is really subtle. If both `releasever` is defined but `source_root` is explicitly passed, then the latter should have precedence, right?